Add glossary for SSG (#35511)
mdn/content: files/en-us/glossary/ssg/index.md

---
title: SSG
slug: Glossary/SSG
page-type: glossary-definition
---

{{GlossarySidebar}}

> A **static site generator** (SSG) is a software used to generate _static_ websites. A static website is comprised of {{glossary("HTML")}}, {{glossary("CSS")}}, and {{glossary("JavaScript")}} files, and most importantly does not have [server-side logic](/en-US/docs/Learn/Server-side), so for any given URL, all users will receive the same content. The developer writes pages in any form accepted by the generator, such as [React](/en-US/docs/Learn/Tools_and_testing/Client-side_JavaScript_frameworks/React_getting_started), Markdown, or HTML templates, and the generator compiles them into a set of optimized static files that can be rendered by the browser.

Reviewer: Fly by comment - this is true, but I don't love the use of the word "developers" here. Most of the time you have a developer who creates an SSG with some form of template framework, you then have authors that create content written in markdown, resTructured text, or some other markup format. As you say, the generator builds this into optimized static files that can be served from a CDN.

As an aside, there is an exception that breaks the rule here, and perhaps it is because this is not "technically an SSG? I'm thinking of docsify since that dynamically loads and renders markdown at runtime. 

But if a rample

Author: Mmmm, I think you are right that typically people who create content in SSGs are not the same as those people who maintain the site. As maintainer of Docusaurus I mainly interact with the latter, so I assumed "user of SSG === developer".

Docsify still statically generates HTML and JS, except the Markdown is basically loaded as asset that's rendered client-side. It still doesn't require a server. I think that would be a technical detail.

```suggestion
A **static site generator** (SSG) is a software used to generate _static_ websites. A static website is comprised of {{glossary("HTML")}}, {{glossary("CSS")}}, and {{glossary("JavaScript")}} files, and most importantly does not have [server-side logic](/en-US/docs/Learn/Server-side), so for any given URL, all users will receive the same content. The author writes pages in any form accepted by the generator, such as [React](/en-US/docs/Learn/Tools_and_testing/Client-side_JavaScript_frameworks/React_getting_started), Markdown, or HTML templates, and the generator compiles them into a set of optimized static files that can be rendered by the browser.
```

Reviewer: That change works for me, though I'd chop that second sentence a bit if possible. Also make it clear that react etc are far less likely to be used.

```suggestion
A **static site generator** (SSG) is a software used to generate _static_ websites.
A static website is comprised of {{glossary("HTML")}}, {{glossary("CSS")}}, and {{glossary("JavaScript")}} files.
Most importantly does not have [server-side logic](/en-US/docs/Learn/Server-side), so for any given URL, all users will receive the same content.
Authors write content in any form accepted by the generator, such markdown, reStructuredText (and sometimes even HTML, [React](/en-US/docs/Learn/Tools_and_testing/Client-side_JavaScript_frameworks/React_getting_started), and so on), and the generator compiles them into a set of optimized static files that can be rendered by the browser.
```

> Docsify still statically generates HTML and JS, except the Markdown is basically loaded as asset that's rendered client-side. It still doesn't require a server. I think that would be a technical detail.

It's certainly a detail that we don't need to argue here :-)

---

> title: Statis site generator (SSG)

Reviewer: ```suggestion
title: Static site generator (SSG)
```

slug: Glossary/SSG
page-type: glossary-definition
---

{{GlossarySidebar}}

> A **static site generator** (SSG) is a software used to generate _static_ websites. A static website is comprised of {{glossary("HTML")}}, {{glossary("CSS")}}, and {{glossary("JavaScript")}} files. Most importantly SSG sites do not have [server-side logic](/en-US/docs/Learn/Server-side), so for any given URL, all users will receive the same content. Authors write content in any form accepted by the generator, such Markdown, reStructuredText, HTML, (and sometimes even [React](/en-US/docs/Learn/Tools_and_testing/Client-side_JavaScript_frameworks/React_getting_started), and so on), and the generator compiles them into a set of optimized static files that can be rendered by the browser.

Author: ```suggestion
A **static site generator** (SSG) is a software used to generate _static_ websites. A static website is comprised of {{glossary("HTML")}}, {{glossary("CSS")}}, and {{glossary("JavaScript")}} files. Most importantly static sites do not have [server-side logic](/en-US/docs/Learn/Server-side), so for any given URL, all users will receive the same content. Authors write content in any form accepted by the generator, such as Markdown, reStructuredText, HTML, (and sometimes even [React](/en-US/docs/Learn/Tools_and_testing/Client-side_JavaScript_frameworks/React_getting_started), and so on), and the generator compiles them into a set of optimized static files that can be rendered by the browser.
```
